instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java

```diff
@@ -205,12 +219,22 @@ public String getHost() {
     public List<String> getHeaders(String name) {
         List<String> headers = new ArrayList<>();
         try {
-            // HTTP/2 only supports lowercase headers
-            String lowerCaseHeaderName = name.toLowerCase();
-            List<CharSequence> allHeaders = http2Headers.getAll(lowerCaseHeaderName);
-            for (CharSequence header : allHeaders) {
-                headers.add(header.toString());
-            }
+            if (!isGRPCHttp2) {
+				// HTTP/2 only supports lowercase headers
+				String lowerCaseHeaderName = name.toLowerCase();
+				List<CharSequence> allHeaders = http2Headers.getAll(lowerCaseHeaderName);
+				for (CharSequence header : allHeaders) {
+					headers.add(header.toString());
+				} 
+			} else {
+				// GRPC only accepts AsciiString
+				AsciiString asciiName = new AsciiString(name);
+				List<CharSequence> allHeaders = http2Headers.getAll(asciiName);
+				for (CharSequence header : allHeaders) {
```

Comment:
Can the `for()` loop be re-factored out of the if/else` blocks?

Reply:
done